tests/websocket.html

```diff
@@ -3,150 +3,215 @@
+						</div>
+					</fieldset>
+				</form>
+				<form class="pure-form">
+					<fieldset>
+						<div class="pure-g">
+							<div class="pure-u-1-3"><input disabled class="pure-u-23-24" type="text" placeholder="key" id="raw-set-key" value="hello" /></div>
+							<div class="pure-u-1-3"><input disabled class="pure-u-23-24" type="text" placeholder="value" id="raw-set-value" value="world" /></div>
+							<div class="pure-u-1-3"><button disabled type="submit" class="pure-u-23-24 pure-button pure-button-primary" id="raw-btn-set">SET</button></div>
+						</div>
+					</fieldset>
+				</form>
+				<form class="pure-form">
+					<fieldset>
+						<div class="pure-g">
+							<div class="pure-u-1-3">&nbsp;</div>
+							<div class="pure-u-1-3"><input disabled class="pure-u-23-24" type="text" placeholder="key" id="raw-get-key" value="hello" /></div>
+							<div class="pure-u-1-3"><button disabled type="submit" class="pure-u-23-24 pure-button pure-button-primary" id="raw-btn-get">GET</button></div>
+						</div>
+					</fieldset>
+				</form>
+
+				<div class="pure-g">
+					<div class="pure-u-2-3">&nbsp;</div>
+					<div class="pure-u-1-3"><button disabled type="submit" class="pure-u-23-24 pure-button pure-button-primary" id="raw-btn-clear">Clear logs</button></div>
+					<div class="log pure-u-1-1" id="raw-log">
+					</div>
+				</div>
 			</div>
-		</section>
 
+			<div class="pure-u-1-8"></div>
+		</div>
 
-		<script type="text/javascript">
+<script type="text/javascript">
 
 $ = function(id) {return document.getElementById(id);};
-var host = "127.0.0.1";
-var port = 7379;
-
-function log(id, dir, msg) {
-
-	var desc = document.createElement("div");
-	desc.setAttribute("class", "desc");
-	desc.innerHTML = dir;
-	$(id).appendChild(desc);
-
-	var e = document.createElement("pre");
-	e.setAttribute("class", dir);
-	e.innerHTML = msg;
-	$(id).appendChild(e);
-}
-
-function testJSON() {
-
-	if(typeof(WebSocket) == 'function')
-		f = WebSocket;
-	if(typeof(MozWebSocket) == 'function')
-		f = MozWebSocket;
-
-	var jsonSocket = new f("ws://"+host+":"+port+"/.json");
-	var self = this;
-
-	send = function(j) {
-		var json = JSON.stringify(j);
-		jsonSocket.send(json);
-		log("json-log", "sent", json);
-	};
-
-	jsonSocket.onopen = function() {
-		$("json-log").innerHTML = "";
-		self.send(["SET", "hello", "world"]);
-		self.send(["GET", "hello"]);
-	};
-
-	jsonSocket.onmessage = function(messageEvent) {
-		log("json-log", "received", messageEvent.data);
-	};
+const host = "127.0.0.1";
+const port = 7379;
+
+class Client {
+	constructor(type, getSerializer, setSerializer) {
+		this.type = type;
+		this.getSerializer = getSerializer;
+		this.setSerializer = setSerializer;
+		this.ws = null;
+
+		$(`${this.type}-btn-connect`).addEventListener('click', event => {
+			event.preventDefault();
+			console.log('Connecting...');
+			this.ws = new WebSocket(`ws://${ host }:${ port }/.${ this.type }`);
```

Comment:
I assume that `MozWebSocket` is no longer a thing?

Reply:
Firefox 90:
![image](https://user-images.githubusercontent.com/61305023/126074368-b1ce7d99-ff03-4f80-8f3a-0bbaabb17fbc.png)